packages/esm-patient-forms-app/src/forms/forms-see-all.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import dayjs from 'dayjs';
+import first from 'lodash-es/first';
+import ContentSwitcher from 'carbon-components-react/es/components/ContentSwitcher';
+import Switch from 'carbon-components-react/es/components/Switch';
+import FormView from './form-view-see-all.component';
+import styles from './forms.component.scss';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { fetchAllForms, fetchPatientEncounters } from './forms.resource';
+import { filterAvailableAndCompletedForms } from './forms-utils';
+import { Encounter, Form } from '../types';
+import EmptyFormView from './empty-form.component';
+import { getObjectFHIR } from './get-FHIR-object';
+
+enum FormViewState {
+  recommended = 0,
+  completed,
+  all,
+}
+
+const Forms = () => {
+  const urlPathArray = window.location.pathname.split('/');
+  const pathUuid = urlPathArray[urlPathArray.length - 2];
+  const [patientUuid, setUUID] = React.useState(null);
+  const [patient, setPatient] = React.useState<fhir.Patient>();
+
+  React.useEffect(() => {
+    getObjectFHIR(patientUuid).then(setPatient);
+    setUUID(pathUuid);
+  }, [pathUuid, patientUuid]);
+
+  const { t } = useTranslation();
+  const displayText = t('forms', 'Forms');
+  const headerTitle = t('forms', 'Forms');
+  const [error, setError] = React.useState(null);
+  const [forms, setForms] = React.useState<Array<Form>>([]);
+  const [encounters, setEncounters] = React.useState<Array<Encounter>>([]);
+  const [completedForms, setCompletedForms] = React.useState<Array<Form>>([]);
+  const [selectedFormView, setSelectedFormView] = React.useState<FormViewState>(FormViewState.all);
+  const [filledForms, setFilledForms] = React.useState<Array<Form>>([]);
+
+  React.useEffect(() => {
+    fetchAllForms().subscribe((forms) => setForms(forms), setError);
+  }, []);
+
+  React.useEffect(() => {
+    const fromDate = dayjs(new Date()).startOf('day').subtract(500, 'day');
+    const toDate = dayjs(new Date()).endOf('day');
+    fetchPatientEncounters(patientUuid, fromDate.toDate(), toDate.toDate()).subscribe(
+      (encounters) => setEncounters(encounters),
+      setError,
+    );
+  }, [patientUuid]);
+
+  React.useEffect(() => {
```

Comment:
To be honest looks more like a `useMemo` to me.

Reply:
This file is a adaptation of forms.component.tsx and I have only made minor alterations to it to fix the small bug and have not tried to create the widget from scratch. Most of the code directly comes from the original file which was used to build the Form Entry widget and is now a part of the "openmrs-esm-patient-chart" repo.

Reply:
We should reuse components - not copy them. This is a no go that results in barely maintainable code that is bloated and contains bugs.

Reply:
Exactly my point here! It was impossible to reuse components here. Let me explain the origins of this bug. If you go to form-view.component.tsx and observe Ln #169, the variable "pageUrl" has been hardcoded. Then again on Ln #170 the variable "pageSize" has been hardcoded. This was the key component required in this situation and it was impossible re-use it. So I had to go for the closest work around. The other option is rewrting everything which is clearly beyond the scope of this ticket. Now, we need to understand why reusable components with hard coded variables are part of our repo in the first place and make corrections there.

Reply:
But putting the old codes aside, I will rewrite certain aspects. I will make changes based on your suggestions and come up with a new commit. Thank you.

Reply:
I have made other changes but probably "useMemo" is not needed here as the fuction gets called only once. Please let me know if I am missing something.

Reply:
Yes you are missing that the `setCompletedForms` is causing a re-render.

Instead, use just a `useMemo`.

Your pattern (bad practice):

```js
const [state, setState] = React.useState();
React.useEffect(() => {
  // long computation giving "result"
  setState(result);
}, []);
```

Instead we can do:

```js
const state = React.useMemo(() => {
  // long computation giving "result"
  return result;
}, []);
```

much more efficient and it's clear from context what goes on...

Reply:
Thanks for explaining. My observation was that even when `setCompletedForms` gets called enforcing a re-render the below block of code does not get executed because the dependacy array will prevent its execution:

`useEffect(() => {
    const fromDate = dayjs(new Date()).startOf('day').subtract(500, 'day');
    const toDate = dayjs(new Date()).endOf('day');
    fetchPatientEncounters(patientUuid, fromDate.toDate(), toDate.toDate()).then(setEncounters);
  }, [patientUuid]);`

I tried replacing this block with `useMemo` but it was not working probably because there is a `fetch` call happening from within this function and the useMemo fails to re-assign the value returned by the "async"  `fetch` function. But there is no problem when I use `useState` functionality. I will try implementing "useMemo" again if you want.

Reply:
> I tried replacing this block with useMemo but it was not working probably because there is a fetch call happening from within this function and the useMemo fails to re-assign the value returned by the "async" fetch function.

Yes, that is a good point. I did not see this beforehand, but the operation is async. I'd still say the code there can be improved (that I could not immediately see there having a resource being fetched just confirms this), but either way - let's leave it as is right now. Thanks for trying!